dali/dfu/dfuutil.cpp

```diff
@@ -858,9 +858,38 @@ class CDFUhelper: public CInterface, implements IDFUhelper
                 throwError1(DFUERR_DFileNotFound, lfn);
         }
         else {
-            Owned<IFileDescriptor> fdesc = file->getFileDescriptor();
-            Owned<IPropertyTree> t = fdesc->getFileTree();
-            toXML(t, out, true);
+            CDfsLogicalFileName tmplfn;
+            StringBuffer lfnpath;
+            bool safe = false;
+            if ((strstr(lfn, "::") != NULL) && !strchr(lfn, '/'))
```

Comment:
I am not sure why you are checking this, it looks like you are seeing whether this looks like a logical file or an xpath?
..it looks like you use it as an xpath if it doesn't look like a logical file below.

* If lfn was e.g. "myfile" - it would not have "::" in it, so it would try to connect to xpath "myfile".
Which is incorrect, previously it would lookup file "myfile" and succeed and use it to export.
* CDFUHelper::getFileXML() expected a logical file name or a group name before [the group bit is a bit bizarre, I don't think it's used anywhere].

It was using DFS interface to get XML, should continue to do, instead of going direct to Dali.
IDistributedFileDirectory::getFileTree() , can export the file xml, whether a plain file or super.